packages/format-library/src/link/inline.js

```diff
@@ -77,6 +77,39 @@ const LinkViewerUrl = ( { url } ) => {
 	);
 };
 
+const URLPopoverAtLink = ( { isActive, addingLink, value, ...props } ) => {
+	const anchorRect = useMemo( () => {
+		const range = window.getSelection().getRangeAt( 0 );
+		if ( ! range ) {
+			return;
+		}
+
+		if ( addingLink ) {
+			return getRectangleFromRange( range );
+		}
+
+		let element = range.startContainer;
+
+		// If the caret is right before the element, select the next element.
+		element = element.nextElementSibling || element;
+
+		while ( element.nodeType !== window.Node.ELEMENT_NODE ) {
+			element = element.parentNode;
+		}
+
+		const closest = element.closest( 'a' );
+		if ( closest ) {
+			return closest.getBoundingClientRect();
+		}
+	}, [ isActive, addingLink, value.start, value.end ] );
```

Comment:
I don't know that `isActive` needed to be listed as a dependency. It's not ever used in the `useMemo` callback.